src/NLog/Internal/NameNormalizer.cs

```diff
@@ -0,0 +1,63 @@
+// 
+// Copyright (c) 2004-2021 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.Internal
+{
+    internal static class NameNormalizer
+    {
+        internal static string NormalizeName(string itemName)
```

Comment:
Would prefer that internal helper methods, that are only used at one place, stays at that place (instead of being extracted into isolated source-file). But I guess you have your reasons.

Reply:
The reason is unit testing. A static method inside a generic class is a bit hard to test.

Reply:
Yes I see, guess that resolves it.

Reply:
After further considering, then I guess it would be better with unit-testing of the actual factory-method. Instead of unit-testing an internal-method.

Reply:
That isn't an unit test but a component test? (Slower, more test code and more maintenance)

Reply:
For NLog is a component, and a tester doing black-box testing of the NLog.dll would be a component-test. I like unit-test to verify an entire contract (here create an object based on type-name),

Reply:
I'm following this: https://martinfowler.com/articles/practical-test-pyramid.html

But I could move the static method I guess

Reply:
move the method

Reply:
If speed is your concern that I think it is just as fast to test the factory directly, as to testing the internal-method. And think the correct way is to test the factory-methods, so it is ensured that the factory methods is using the internal-method correctly.